gym-unity/gym_unity/envs/__init__.py

```diff
@@ -344,6 +358,79 @@ def _check_agents(self, n_agents: int) -> None:
                 "initialization. This is not supported."
             )
 
+    def _sanitize_info(self, info: BatchedStepResult) -> BatchedStepResult:
+        self._stored_info = info  # store the new original
+        if info.n_agents() == self._n_agents:
+            return info
+        n_extra_agents = info.n_agents() - self._n_agents
+        if self._n_agents is None:
+            self._n_agents = 1
+        if n_extra_agents < 0 or n_extra_agents > 2 * self._n_agents:
```

Comment:
What about Agents that don't make a decision every step? Wouldn't that lead to `info.n_agents() < self._n_agents`?

What about an Agent that calls Done() multiple times in the same step? Wouldn't that lead to `info.n_agents() > 2 * self._n_agents`

Reply:
Yes, but you should not call Done multiple times per step : )

Reply:
Then we should assert that on the C# side.

Reply:
And/or provide a more descriptive error message here

Reply:
If Done is sent multiple times, the gym will ignore it if it is sent between decision steps (this is what the while loop is for)